File: test-junit5/src/test/java/io/micronaut/test/junit5/PropertySourceTest.java

package io.micronaut.test.junit5;

import io.micronaut.context.annotation.Property;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

@MicronautTest(propertySources = "myprops.properties")
class PropertySourceTest {

    @Property(name = "foo.bar")
    String val;


    @Test
    void testPropertySource() {
        Assertions.assertEquals("foo", val);
    }
}

@MicronautTest(propertySources = "file:src/test/resources/io/micronaut/test/junit5/fileprops.properties")
class FilePropertySourceTest {

    @Property(name = "foo.file")
    String val;

    @Test
    void testFilePropertySource() {
        Assertions.assertEquals("file", val);
    }
}

File: test-suite-python/src/test/python/micronaut/test/python/PropertySourceTest.py
from typing import Annotated

from micronaut.context.annotation import Property
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@MicronautTest(propertySources="myprops.properties")
class PropertySourceTest:

    val: Annotated[str, Property(name="foo.bar")]

    @Test
    def test_property_source(self):
        assert "foo" == self.val


@MicronautTest(propertySources="file:src/test/resources/micronaut/test/python/fileprops.properties")
class FilePropertySourceTest:

    val: Annotated[str, Property(name="foo.file")]

    @Test
    def test_file_property_source(self):
        assert "file" == self.val
